Dockerfile.local

```diff
@@ -0,0 +1,9 @@
+FROM gcr.io/distroless/static:debug
+
+WORKDIR /
+COPY bin/k8s .
+COPY bin/unpack .
+EXPOSE 8080
+
+ENTRYPOINT ["/k8s"]
+CMD ["run"]
```

Comment:
Do we need a separate Dockerfile? It seems like that could lead to local development behaviors diverging from CI/release image behaviors.

What if we change the existing Dockerfile to copy in a local binary and update the make targets that build it to:
1. Enforce a specific go version
2. Ensure the underlying binary is rebuilt before running `docker build`

Reply:
The existing Dockerfile is a multi-phase docker file.  It setup the binary build environment, build binaries and creates the container image.  It doesn't depend on the environment of the system.  It can build the consistent image.  The Dockerfile.local has just container built part.  The executable builds are depend of the local system.  It must be setup correctly but it builds the binaries faster than the building in the docker build process.

Reply:
Yeah, I'm asserting that the enforcement of a specific Go version is really the only system environment thing that matters. 

I'm making that assertion based on the fact that the popular GoReleaser project builds images by building binaries locally and then copying them into an image: https://goreleaser.com/customization/docker/

Reply:
Interested in @timflannagan and @exdx thoughts on this idea.

Reply:
Honestly I think we just punt on consolidating any CI vs. production Dockerfile situation right now, and enable the ability to build binaries locally, copy those binaries into a dockerfile, and load that image to a kind cluster to help enable dev workflows but maybe I'm biased? The only consideration that comes into mind is whether this kind of setup plays well with loading images to kind node(s) if docker/podman/etc. aren't able to detect new digest images, and if that's the case, then this kind of setup is absolutely watered down being able to quickly iterate without pushing these build dev container images to a remote container registry.

Reply:
> enable the ability to build binaries locally, copy those binaries into a dockerfile

Big +1 from me. I'm saying that's all we need. Let's use that same setup in CI.

> docker/podman/etc. aren't able to detect new digest images

This flow should work. Though I think we would need to `uninstall` and then `install` to cycle the deployment.

Reply:
Both ways have pros and cons.  I think that providing make targets and docker files for both way is good at this stage.

Reply:
I'm also ok with having two separate Dockerfiles at this stage but if @joelanford feels strongly we should consolidate then we could update this PR to remove the existing Dockerfile as well. 

My impression is that this PR was meant to be purely additive though and just provide a local Dockerfile. We could reassess our deployment strategy in a follow-up.

Reply:
> Yeah, I'm asserting that the enforcement of a specific Go version is really the only system environment thing that matters.

I'm wondering how we could enforce that requirement - just have a script that runs `go version` before building and checks that the output is what we expect?

Reply:
My only thought is that one make target + one Dockerfile is easier to maintain and reason about than two, but I don't have a strong enough feeling to block progress.

>  runs go version before building and checks that the output is what we expect

Yep, pretty much.

Reply:
I just took another look at this lgtm. It sounds like there's reservations around potential skew between CI/local Dockerfiles, and we may want to consolidate this over time (which Joe has already pointed put here). Can we open an issue to track this concern so we don't lose this context over time, and push this through? @akihikokuroda would you mind creating this issue, and then we can merge this?